packages/nx/src/tasks-runner/run-command.ts

```diff
@@ -571,12 +600,17 @@ export async function invokeTasksRunner({
     taskGraph,
     nxJson
   );
+  const compositedLifeCycle: LifeCycle = new CompositeLifeCycle(
+    constructLifeCycles(lifeCycle)
+  );
```

Comment:
This can be:

```suggestion
  const taskResultsLifecycle = new TaskResultsLifeCycle();
  const compositedLifeCycle: LifeCycle = new CompositeLifeCycle([
    ...constructLifeCycles(lifeCycle),
    taskResultsLifecycle
  ]);
```